tests/test_extensions/test_ext_autosummary.py

```diff
@@ -488,6 +488,376 @@ def test_autosummary_generate_content_for_module_imported_members_inherited_modu
+            'autosummary_dummy_module.Foo.CONSTANT3',
+            'autosummary_dummy_module.Foo.CONSTANT4',
+            'autosummary_dummy_module.Foo.bar',
+            'autosummary_dummy_module.Foo.baz',
+            'autosummary_dummy_module.Foo.value',
+        ],
+        context['inherited_qualnames'],
+    )
+    assert context['inherited_methods'] == ['autosummary_dummy_module.Foo.bar']
+    assert context['inherited_attributes'] == [
+        'autosummary_dummy_module.Foo.CONSTANT3',
+        'autosummary_dummy_module.Foo.CONSTANT4',
+        'autosummary_dummy_module.Foo.baz',
+        'autosummary_dummy_module.Foo.value',
+    ]
+    assert (
+        context['fullname'] == 'autosummary_dummy_inherited_module.InheritedAttrClass'
+    )
+    assert context['module'] == 'autosummary_dummy_inherited_module'
+    assert context['objname'] == 'InheritedAttrClass'
+    assert context['name'] == 'InheritedAttrClass'
+    assert context['objtype'] == 'class'
+
+
+@pytest.mark.sphinx(testroot='ext-autosummary')
+def test_autosummary_generate_content_for_module_imported_members_complex_inheritance(
+    app,
+):
+    import autosummary_dummy_complex_inheritance_module
+
+    built_in_attr = autosummary_dummy_complex_inheritance_module.built_in_attr
+    built_in_members = autosummary_dummy_complex_inheritance_module.built_in_members
+    built_in_members2 = autosummary_dummy_complex_inheritance_module.built_in_members2
+
+    template_jerry = Mock()
+
+    generate_autosummary_content(
+        name='autosummary_dummy_complex_inheritance_module.Jerry',
+        obj=autosummary_dummy_complex_inheritance_module.Jerry,
+        parent=None,
+        template=template_jerry,
+        template_name=None,
+        imported_members=True,
+        app=app,
+        recursive=False,
+        context={},
+    )
+    assert template_jerry.render.call_args[0][0] == 'class'
+
+    context = template_jerry.render.call_args[0][1]
+
+    assert context['name'] == 'Jerry'
+    assert context['module'] == 'autosummary_dummy_complex_inheritance_module'
+    assert context['fullname'] == 'autosummary_dummy_complex_inheritance_module.Jerry'
+
+    assert context['attributes'] == ['relation']
+    assert context['methods'] == [
+        '__init__',
+        'addition',
+        'get_age',
+        'get_name',
+        'get_salary',
+    ]
+
+    assert context['all_attributes'] == [
+        *built_in_attr,
+        'relation',
+    ]
+    assert context['all_methods'] == [
+        '__delattr__',
+        '__dir__',
+        '__eq__',
+        '__format__',
+        '__ge__',
+        '__getattribute__',
+        '__getstate__',
+        '__gt__',
+        '__hash__',
+        '__init__',
+        '__init_subclass__',
+        '__le__',
+        '__lt__',
+        '__ne__',
+        '__new__',
+        '__reduce__',
+        '__reduce_ex__',
+        '__repr__',
+        '__setattr__',
+        '__sizeof__',
+        '__str__',
+        '__subclasshook__',
```

Comment:
Aren't those methods inherited from `object`? can't they be automatically detected like this?